src/main/java/com/faforever/client/vault/VaultEntityController.java

```diff
@@ -327,20 +325,7 @@ protected void onBackButtonClicked() {
   protected void onDisplay(NavigateEvent navigateEvent) {
     Class<? extends NavigateEvent> defaultNavigateEvent = getDefaultNavigateEvent();
     if (!(navigateEvent.getClass().equals(defaultNavigateEvent)) && !navigateEvent.getClass().equals(NavigateEvent.class)) {
-      if (state.get() == State.UNINITIALIZED) {
-        state.addListener(new ChangeListener<>() {
-          @Override
-          public void changed(ObservableValue<? extends State> observable, State oldValue, State newValue) {
-            if (newValue != State.UNINITIALIZED) {
-              state.removeListener(this);
-              handleSpecialNavigateEvent(navigateEvent);
-            }
-          }
-        });
-        loadShowRoom();
-      } else {
-        handleSpecialNavigateEvent(navigateEvent);
-      }
```

Comment:
Well I think you fixed it but now if you click the back button after the search it showroom is empty. That is why this code existed. Maybe easier would be check if state is uninitialized on back button click and call loadShowRoom then

Reply:
I don't get this error the show room always populates for me with the back button. How are you getting an empty showroom?

Reply:
In the code the back button always loads the showroom

Reply:
Was just an assumption because in the old code that used to be the case

Reply:
yeah if you test it and it happens let me know cause it isnt happening for me